apps/omg_watcher/test/exit_processor/core_test.exs

```diff
@@ -766,6 +806,330 @@ defmodule OMG.Watcher.ExitProcessor.CoreTest do
+      request =
+        exit_processor_request
+        |> struct!(%{piggybacked_utxos_to_check: [Utxo.position(1, 0, 0)], piggybacked_utxo_exists_result: [false]})
+        |> Core.determine_ife_spends_to_get(processor)
+
+      assert Utxo.position(1, 0, 0) in request.piggybacked_spends_to_get
+
+      assert %ExitProcessor.Request{piggybacked_blknums_to_get: [1]} =
+               exit_processor_request
+               |> struct!(%{piggybacked_spends_to_get: [Utxo.position(1, 0, 0)], piggybacked_spent_blknum_result: [1]})
+               |> Core.determine_ife_blocks_to_get()
+    end
+
+    @tag fixtures: [:alice, :processor_filled, :transactions, :in_flight_exits, :competing_transactions]
+    test "detects multiple double-spends in single IFE",
+         %{
+           alice: alice,
+           processor_filled: state,
+           transactions: [tx | _],
+           competing_transactions: [comp | _],
+           in_flight_exits: [{ife_id, _} | _]
+         } do
+      txbytes = Transaction.encode(tx)
+      comp_txbytes = Transaction.encode(comp)
+
+      {:ok, recovered} = DevCrypto.sign(tx, [alice.priv, alice.priv]) |> Transaction.Recovered.recover_from()
+      %{sigs: [_, other_signature]} = DevCrypto.sign(comp, [<<>>, alice.priv])
+
+      other_ife_event = %{call_data: %{in_flight_tx: comp_txbytes, in_flight_tx_sigs: other_signature}, eth_height: 2}
+      other_ife_status = {1, <<1::192>>}
+
+      {state, _} = Core.new_in_flight_exits(state, [other_ife_event], [other_ife_status])
+
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 0}])
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 1}])
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 4}])
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 5}])
+
+      tx_blknum = 3000
+
+      {request, state} =
+        %ExitProcessor.Request{
+          blknum_now: 4000,
+          eth_height_now: 5,
+          piggybacked_blocks_result: [Block.hashed_txs_at([recovered], tx_blknum)]
+        }
+        |> Core.find_ifes_in_blocks(state)
+
+      # note that outputs are not double-spent and can't be challenged; they can't exit since
+      # IFE is not canonical
+      assert {:ok, [%Event.InvalidPiggyback{txbytes: ^txbytes, inputs: [0], outputs: []}]} =
+               invalid_exits_filtered(request, state, only: [Event.InvalidPiggyback])
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "produces single challenge proof on double-spent piggyback input",
+         %{
+           invalid_piggyback_on_input: %{
+             state: state,
+             request: request,
+             bad_pb_output: bad_pb_output,
+             ife_txbytes: txbytes
+           }
+         } do
+      assert {:ok, [%{in_flight_input_index: ^bad_pb_output}]} =
```

Comment:
assert all fields, similar to competitors tests etc. Are there any other behaviors related to `get_input_challenge_data` which we want to test?

Reply:
done